zerver/tests/test_subs.py

```diff
@@ -823,6 +826,158 @@ def test_api_calls(self):
         self.assert_json_success(result)
         self.assertFalse(stream_name in self.get_default_stream_names(user_profile.realm))
 
+class DefaultStreamGroupTest(ZulipTestCase):
+    def test_add_update_and_remove_default_stream_group(self):
+        # type: () -> None
+        realm = get_realm("zulip")
+        default_stream_groups = get_default_stream_groups_for_realm(realm)
+        self.assert_length(default_stream_groups, 0)
+
+        stream_names = ["stream1", "stream2", "stream3"]
+        streams = []
+
+        for stream_name in stream_names:
+            (stream, _) = create_stream_if_needed(realm, stream_name)
+            streams.append(stream)
+
+        group_name = "group1"
+
+        do_create_default_stream_group(realm, group_name, streams)
+        default_stream_groups = get_default_stream_groups_for_realm(realm)
+        self.assert_length(default_stream_groups, 1)
+        self.assertEqual(default_stream_groups[0].name, group_name)
+        self.assertEqual(list(default_stream_groups[0].streams.all()), streams)
+
+        new_stream_names = ["stream4", "stream5"]
+        new_streams = []
+        for new_stream_name in new_stream_names:
+            (new_stream, _) = create_stream_if_needed(realm, new_stream_name)
+            new_streams.append(new_stream)
+            streams.append(new_stream)
+
+        do_add_streams_to_default_stream_group(realm, group_name, new_streams)
+        default_stream_groups = get_default_stream_groups_for_realm(realm)
+        self.assert_length(default_stream_groups, 1)
+        self.assertEqual(default_stream_groups[0].name, group_name)
+        self.assertEqual(list(default_stream_groups[0].streams.all()), streams)
+
+        do_remove_streams_from_default_stream_group(realm, group_name, new_streams)
+        default_stream_groups = get_default_stream_groups_for_realm(realm)
+        self.assert_length(default_stream_groups, 1)
+        self.assertEqual(default_stream_groups[0].name, group_name)
+        del streams[3:5]
```

Comment:
I might just set `streams` more explicitly or add a comment; this `del` statement works, but feels like it could be confusing.